twittermap/public/javascripts/cache/services.js

```diff
@@ -0,0 +1,321 @@
+
+       var deferred = new $.Deferred();
+       var cache_bbox = turf.bbox(cachedRegion);
+       var C_minX = cache_bbox[0];
+       var C_minY = cache_bbox[1];
+       var C_maxX = cache_bbox[2];
+       var C_maxY = cache_bbox[3];
+
+       var request_bbox = turf.bbox(currentRequest);
+       var R_minX = request_bbox[0];
+       var R_minY = request_bbox[1];
+       var R_maxX = request_bbox[2];
+       var R_maxY = request_bbox[3];
+
+       var CacheMBR = turf.bboxPolygon(cache_bbox);
+       //Four Corners of Request to see which is inside cache_region
+       var UpperRight = turf.inside(turf.point([R_maxX,R_maxY]),CacheMBR);
+       var UpperLeft  = turf.inside(turf.point([R_minX,R_maxY]),CacheMBR);
+       var LowerLeft  = turf.inside(turf.point([R_minX,R_minY]),CacheMBR);
+       var LowerRight = turf.inside(turf.point([R_maxY,R_minY]),CacheMBR);
+
+
+       if(LowerRight || LowerLeft && !UpperRight && !UpperLeft){
+            //Y from bottom to top
+            cutRegion(C_minX,C_minY,C_maxX,R_minY,false,true) .done(function(){
+                    deferred.resolve();
+            }).fail(function(){
+
+                     clearCache().done();
+                     deferred.resolve();
+            })
+            return deferred.promise();
+       }
+       else if(UpperRight || UpperLeft && !LowerLeft && !LowerRight){
+            cutRegion(C_minX,R_maxY,C_maxX,C_maxY,false,false).done(function(){
+            //Y from top to bottom
+                   deferred.resolve();
+            }).fail(function(){
+
+                    clearCache().done();
+                    deferred.resolve();
+            })
+            return deferred.promise();
+
+       }
+       else if(UpperRight && LowerRight){
+            cutRegion(R_maxX,C_minY,C_maxX,C_maxY,true,true).done(function(){
+            //X from left to right
+                   deferred.resolve();
+            }).fail(function(){
+
+                    clearCache().done();
+                    deferred.resolve();
+            })
+            return deferred.promise();
+
+       }else if(UpperLeft && LowerLeft){
+            cutRegion(C_minX,C_minY,R_minX,C_maxY,true,false).done(function(){
+            //X from right to left
+                   deferred.resolve();
+            }).fail(function(){
+
+                    clearCache().done();
+                    deferred.resolve();
+            })
+            return deferred.promise();
+
+       }else if(!LowerRight &&   !LowerLeft && !UpperLeft && !UpperRight){
+
+              cutRegion(C_minX,C_minY,C_maxX,C_maxY,true,true).done(function(){
+              //NO Overlap
+                     deferred.resolve();
+              }).fail(function(){
+
+                      clearCache().done();
+                      deferred.resolve();
+              })
+              return deferred.promise();
+       }
+ }
+//sees whether cutting the region can satisfy the target ,If not sends a failed message to evict function
```

Comment:
Please rephrase the comment: "Checks whether evicting some city polygons in a part of the cache region satisfies the target deletion count. If not, returns a failure message to the caller."